microcosm_eventsource/factory.py

```diff
@@ -101,6 +103,13 @@ def validate_required_fields(self, event_info, **kwargs):
                 ],
             )
 
+    def validate_additional(self, event_info, **kwargs):
```

Comment:
Can you name this more specifically? We've already validated the required fields and the transition logic; what kind of decisions to you expect to happen here?

Reply:
How about `validate_custom_implementation`? Just to support logic around what is in: _validate_operate_on_version in the https://github.com/globality-corp/prudencio/pull/73/files PR.